elastic/kibana
# Allow IdP initiated SAML login with session containing expired token.

As in proposed in https://github.com/elastic/kibana/issues/59629 in this PR we're trying to leverage existing session during IdP initiated SAML login even if it contains expired access token.

**Notable changes:**

* We try to use SAML Response even if existing access token is expired
* We treat `404` errors that Elasticsearch returns when it tries to invalidate non-existent tokens in exactly the same way as we'd treat `404` response (see https://github.com/elastic/elasticsearch/pull/54532#issue-396571241 for details)

**How to test:**

1. Log in with SAML (via IdP or User initiated flow, it doesn't matter)
2. Delete all access tokens from Elasticsearch:
```http
POST localhost:9200/.security-tokens/_delete_by_query?pretty&refresh=true
Authorization: Basic xxxx
Accept: application/json
Content-Type: application/json

{
  "query": {
    "match": {
      "doc_type": "token"
    }
  }
}
```
3. Try to log in to Kibana via IdP initiated flow using the same or another user __in the same browsing context__ (so that browser sends SAML Response with the existing cookie with deleted tokens)

Note: You may want to close browser tab (not window) after step 1 so that Kibana doesn't automatically log you out. 

~~__Blocked by:__ https://github.com/elastic/elasticsearch/issues/53323~~
__Fixes:__ https://github.com/elastic/kibana/issues/59629

-------------------------------

"Release Note: previously user couldn't log in with SAML Identity Provider Initiated flow (e.g. from Okta Dashboard) if they already had an existing, but expired session. Now it should be possible."
